src/org/sosy_lab/java_smt/test/SolverContextFactoryTest.java

```diff
@@ -119,20 +122,38 @@ private void requireSupportedOperatingSystem() {
     }
   }
 
-  /** Some libraries require GLIBCXX in version 3.4.26 or newer. This excludes Ubuntu 18.04. */
+  /**
+   * Some libraries require GLIBC in version 2.34 or GLIBCXX in version 3.4.26 or newer. This
+   * excludes Ubuntu 18.04 or 20.04 for some solvers.
+   */
   private boolean isSufficientVersionOfLibcxx(String library) {
     try {
       NativeLibraries.loadLibrary(library);
     } catch (UnsatisfiedLinkError e) {
-      for (String version : new String[] {"3.4.26", "3.4.29"}) {
-        if (e.getMessage().contains("version `GLIBCXX_" + version + "' not found")) {
+      for (String dependency : getRequiredLibcxx(library)) {
+        if (e.getMessage().contains("version `" + dependency + "' not found")) {
           return false;
         }
       }
     }
     return true;
   }
 
+  private String[] getRequiredLibcxx(String library) {
```

Comment:
This is very nice!
We should add this information (and requirements in general) to the README.

Reply:
Good point. I will update the Linux-column in the README with this information.

Overall, we might think about dropping support for anything older than Ubuntu 22.04 (`GLIBC_2.33` and `GLIBCXX_3.4.26`), because the overhead in developement effort is quite high for quite few users.

Reply:
Thank you!

I agree with that. Maybe we should release one more version with minimal changes, e.g. 5.1, and add all of the coming changes in a way that users can prepare.

Reply:
Readme is updated now.